wazimap_ng/points/services/locations.py

```diff
@@ -2,21 +2,23 @@
 
 from wazimap_ng.points.models import Location, Category, ProfileCategory
 from wazimap_ng.profile.models import Profile
-from wazimap_ng.datasets.models import Geography
+from wazimap_ng.datasets.models import Geography, Version
 from wazimap_ng.boundaries.models import GeographyBoundary
 
 logger = logging.getLogger(__name__)
 
-def get_locations(queryset, profile, category=None, geography_code=None):
+def get_locations(
+        queryset, profile, category=None, geography_code=None, version_name=None
+    ):
     geography = None
 
     if category is not None:
         queryset = queryset.filter(category=category)
 
     if geography_code is not None:
-        version = profile.geography_hierarchy.root_geography.version
-        geography = Geography.objects.get(code=geography_code, version=version) 
-        boundary = GeographyBoundary.objects.get(geography=geography)
+        version = Version.objects.get(name=version_name)
+        boundary = GeographyBoundary.objects.get(geography__code=geography_code, version=version)
```

Comment:
Might throw error if:
* Geo code does not exists. We are not checking if geo exist in views
* will throw error if boundary does not exist for version and geo code